Formula/onlykey-agent.rb

```diff
@@ -0,0 +1,219 @@
+    sha256 "3c77e014170dfffbd816e6ffc205e9842efb10be9f58ec16d3e8675b4925cddb"
+  end
+
+  resource "python-daemon" do
+    url "https://files.pythonhosted.org/packages/d6/2d/f5e9a44e76777ca5c20765d92eb2d2cb89b6cfa5e921c808fdd57c90cae7/python-daemon-2.3.0.tar.gz"
+    sha256 "bda993f1623b1197699716d68d983bb580043cf2b8a66a01274d9b8297b0aeaf"
+  end
+
+  resource "pyusb" do
+    url "https://files.pythonhosted.org/packages/d9/6e/433a5614132576289b8643fe598dd5d51b16e130fd591564be952e15bb45/pyusb-1.2.1.tar.gz"
+    sha256 "a4cc7404a203144754164b8b40994e2849fde1cfff06b08492f12fff9d9de7b9"
+  end
+
+  resource "requests" do
+    url "https://files.pythonhosted.org/packages/e9/23/384d9953bb968731212dc37af87cb75a885dc48e0615bd6a303577c4dc4b/requests-2.28.0.tar.gz"
+    sha256 "d568723a7ebd25875d8d1eaf5dfa068cd2fc8194b2e483d7b1f7c81918dbec6b"
+  end
+
+  resource "semver" do
+    url "https://files.pythonhosted.org/packages/31/a9/b61190916030ee9af83de342e101f192bbb436c59be20a4cb0cdb7256ece/semver-2.13.0.tar.gz"
+    sha256 "fa0fe2722ee1c3f57eac478820c3a5ae2f624af8264cbdf9000c980ff7f75e3f"
+  end
+
+  resource "Unidecode" do
+    url "https://files.pythonhosted.org/packages/41/16/ee78864c2c2ba9bddba3978baa378f83270c8b7810eb7900f84e62a9ffac/Unidecode-1.3.4.tar.gz"
+    sha256 "8e4352fb93d5a735c788110d2e7ac8e8031eb06ccbfe8d324ab71735015f9342"
+  end
+
+  resource "urllib3" do
+    url "https://files.pythonhosted.org/packages/1b/a5/4eab74853625505725cefdf168f48661b2cd04e7843ab836f3f63abf81da/urllib3-1.26.9.tar.gz"
+    sha256 "aabaf16477806a5e1dd19aa41f8c2b7950dd3c746362d7e3223dbe6de6ac448e"
+  end
+
+  resource "wcwidth" do
+    url "https://files.pythonhosted.org/packages/89/38/459b727c381504f361832b9e5ace19966de1a235d73cdbdea91c771a1155/wcwidth-0.2.5.tar.gz"
+    sha256 "c4d647b99872929fdb7bdcaa4fbe7f01413ed3d98077df798530e5b04f116c83"
+  end
+
+  def install
+    # prevent "fatal error: libusb.h: No such file or directory" when building hidapi on linux
+    ENV.append_to_cflags "-I#{Formula["libusb"].include}/libusb-1.0"
+    # replacement for virtualenv_install_with_resources per https://docs.brew.sh/Python-for-Formula-Authors
+    venv = virtualenv_create(libexec, "python3")
+    # build hidapi
+    resource("hidapi").stage do
+      # monkey patch hidapi's include paths to be the homebrew-installed path instead
+      # TODO: Fix this with an upstream patch to support `--with-system-hidapi=/foo/bar/hidapi`
+      #       per https://github.com/Homebrew/homebrew-core/pull/104096#discussion_r919469723
+      inreplace "setup.py" do |s|
+        s.gsub! "/usr/include/libusb-1.0", "#{Formula["libusb"].opt_include}/libusb-1.0"
+        s.gsub! "/usr/include/hidapi", "#{Formula["hidapi"].opt_include}/hidapi"
+      end
+      system libexec/"bin/python3", *Language::Python.setup_install_args(libexec), "--with-system-hidapi"
```

Comment:
Does using `pip` to install this work? If so, we may be able to do `venv.pip_install` (which is just a wrapper around `pip install`).

Reply:
@carlocab We need to be able to pass the `--with-system-hidapi` option to this during install. I looked at other formula usages of `.pip_install` and didn't see any examples of that. I also looked at the source for `pip_install` and didn't notice any way to pass such an option. Am I missing something?

Reply:
You are not. I had forgotten that we can't pass extra arguments to `#pip_install`. Thanks for looking into it!